contracts/Holdable.sol

```diff
@@ -0,0 +1,225 @@
+pragma solidity ^0.5.0;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "./IHoldable.sol";
+import "./libraries/StringUtil.sol";
+
+
+contract Holdable is IHoldable, ERC20 {
+
+    using SafeMath for uint256;
+    using StringUtil for string;
+
+
+    struct Hold {
+        address issuer;
+        address origin;
+        address target;
+        address notary;
+        uint256 expiration;
+        uint256 amount;
+        HoldStatusCode status;
+    }
+
+    mapping(string => Hold) private holds;
+    mapping(address => uint256) private heldBalance;
+    mapping(address => mapping(address => bool)) private operators;
+
+    uint256 private _totalHeldBalance;
+
+    function hold(string calldata operationId, address to, address notary, uint256 value, uint256 timeToExpiration) external returns (bool){
+
+        require(holds[operationId].amount == 0 && holds[operationId].target == address(0) && holds[operationId].notary == address(0), "This operationId already exists");
+        require(value <= balanceOf(msg.sender), "Amount of the hold can't be greater than the balance of the origin");
+
+        holds[operationId].issuer = msg.sender;
+        holds[operationId].origin = msg.sender;
+        holds[operationId].target = to;
+        holds[operationId].notary = notary;
+        holds[operationId].amount = value;
+        holds[operationId].status = HoldStatusCode.Ordered;
+
+        if(timeToExpiration == 0){
+            holds[operationId].expiration = 0;
+        }else{
+            holds[operationId].expiration = block.timestamp.add(timeToExpiration);
+        }
+
+        heldBalance[msg.sender] = heldBalance[msg.sender].add(value);
+
+
+        _totalHeldBalance = _totalHeldBalance.add(value);
+
+        emit HoldCreated(
+            msg.sender,
+            operationId,
+            msg.sender,
+            to,
+            notary,
+            value,
+            timeToExpiration
+        );
+        return true;
+    }
+
+
+    function holdFrom(string calldata operationId, address from, address to, address notary, uint256 value, uint256 timeToExpiration) external returns (bool){
+
+        require(holds[operationId].amount == 0 && holds[operationId].target == address(0) && holds[operationId].notary == address(0), "This operationId already exists");
+        require(value <= balanceOf(from), "Amount of the hold can't be greater than the balance of the origin");
+        require (operators[from][msg.sender] == true, "This operator is not authorized");
```

Comment:
` == true` is not needed, because require checks automatically for `true`. So you can write:

```
require (operators[from][msg.sender], "This operator is not authorized");
```